ui-modules/blueprint-composer/app/components/providers/blueprint-service.provider.js

```diff
@@ -512,6 +514,16 @@ function BlueprintService($log, $q, $sce, paletteApi, iconGenerator, dslService)
         entity.miscData.set('config', allConfig);
     }
 
+    function addUnlistedParameterDefinitions(entity) {
+        let allParams = entity.miscData.get('parameters') || [];
+        entity.parameters.forEach((param) => {
+            if (!allParams.some((e) => e.name === param.name)) {
+                allParams.push(param);
+            }
+        });
+        entity.miscData.set('parameters', allParams);
+    }
```

Comment:
it's doing the same as `addUnlistedConfigKeyDefinition` -- although why it is needed I'm not clear, feels like it should be the responsibility of `miscData` to populate those, not this code here.

worth a comment to explain but not needed for this PR.

i say this is good to merge.

Reply:
`addUnlistedConfigKeyDefinition` has the specific function to create the definition of a config key that we add. We don't do that for parameters hence we should remove this method **before merging it**

Reply:
aha TY.  when we add paramters via the UI (coming soon) this will be needed.  okay to add with a comment to this effect in the `addUnlilstedParameters` method?

Reply:
@tbouron noted to me

> Hum but the parameters are already definitions, we just need to add them into the model which Aled did. There is no need to loop over the parameters to add them again

i think that makes sense so he is right, let's remove this method @aledsage -- and be sure to test DSL when we have parameters defined via the UI